recipes/compute_slurm_config.rb

```diff
@@ -35,12 +35,8 @@
   retry_delay 6
 end
 
-# Check to see if there is GPU on the instance, only execute run_nvidiasmi if there is GPU
-if graphic_instance?
-  execute "run_nvidiasmi" do
-    command 'nvidia-smi'
-  end
-end
+# Check to see if there is GPU on the instance, only execute run_nvidiasmi if there is GPU and nvidia installed
```

Comment:
Check for the existence of the binary is what we do when we checking whether to install Nvidia in `nvidia_install.rb` recipe. So I think it is enough. We are not running executing run_nvidiasmi anymore because running it in an instance without Nvidia the chef will failed:
Have been test with the following two cases:
adding an exit code:
```
def nvidia_installed?
  nvidia_installed = Mixlib::ShellOut.new("nvidia-smi").run_command

  Chef::Log.warn("Nvidia drive is not installed") unless nvidia_installed.exitstatus.zero?

  nvidia_installed.exitstatus.zero?
end
```
with error
```
nvidia_installed = Mixlib::ShellOut.new("nvidia-smi").run_command error:
No such file or directory - nvidia-smi
```
Also test with 
```
nvidia_installed = ! shell_out("nvidia-smi").error?
return nvidia_installed
```
It also failed and the process can not continue